classes/class-nets-easy-order-management.php

```diff
@@ -71,8 +71,15 @@ public function dibs_order_completed( $order_id ) {
 				$wc_order->add_order_note( sprintf( __( 'No charge needed in Nexi system since the order total is %s.', 'dibs-easy-for-woocommerce' ), $wc_order->get_total() ) );
 				return;
 			}
+
+			$payment_id = $wc_order->get_meta( '_dibs_payment_id' );
+			if ( empty( $payment_id ) ) {
+				Nets_Easy_Logger::log( 'No payment ID found in the order completed request for WooCommerce order number: ' . $wc_order->get_order_number() );
```

Comment:
I would suggest also using the function we use in the other failed state conditions: `$this->fetching_order_failed( $wc_order, true, 'Your error message' )` since this will also set the order back to on-hold and print a order note on the order letting the merchant know. With how this is, we would log it, if the merchant has logging enabled, but the merchant would see an order that is completed and have no clear indication that something went wrong.